src/StmtToViz.cpp

```diff
@@ -0,0 +1,2900 @@
+#include "StmtToViz.h"
+#include "Debug.h"
+#include "Error.h"
+#include "IROperator.h"
+#include "IRPrinter.h"
+#include "IRVisitor.h"
+#include "Module.h"
+#include "Scope.h"
+#include "Substitute.h"
+#include "Util.h"
+
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <regex>
+#include <sstream>
+#include <utility>
+
+namespace Halide {
+namespace Internal {
+
+using std::istringstream;
+using std::ostringstream;
+using std::string;
+
+extern "C" unsigned char halide_html_template_StmtToViz_dependencies[];
+extern "C" unsigned char halide_html_template_StmtToViz_stylesheet[];
+extern "C" unsigned char halide_html_template_StmtToViz_javascript[];
+
+// Classes define within this file
+class CostModel;
+class AssemblyInfo;
+template<typename T>
+class HTMLCodePrinter;
+class HTMLVisualizationPrinter;
+class IRVisualizer;
+
+/********************** IRCostModel **********************/
+// A basic cost model for Halide IR. Estimates computation
+// cost through simple op-counting and data-movement cost
+// by counting the number of bits being moved.
+class IRCostModel : public IRVisitor {
+public:
+    IRCostModel()
+        : _max_compute_cost(-1), _max_data_cost(-1), _max_compute_cost_inclusive(-1),
```

Comment:
style nit: Halide never uses an underscore-prefix for member variables. It occasionally uses underscore-suffix (e.g. `max_compute_cost_`), but usually only when necessary to avoid ambiguity with another symbol. Preferred style in Halide is just a plain snake-case name (e.g. `max_compute_cost`). Please update here and elsewhere as needed.